ports/zephyr/modbluetooth_zephyr.c

```diff
@@ -56,9 +77,42 @@ enum {
     MP_BLUETOOTH_ZEPHYR_GAP_SCAN_STATE_ACTIVE,
 };
 
+union uuid_u {
+    struct bt_uuid uuid;
+    struct bt_uuid_16 u16;
+    struct bt_uuid_32 u32;
+    struct bt_uuid_128 u128;
+};
+
+struct add_characteristic {
+    uint8_t properties;
+    uint8_t permissions;
+    const struct bt_uuid *uuid;
+};
+
+struct add_descriptor {
+    uint8_t permissions;
+    const struct bt_uuid *uuid;
+};
+
+typedef struct _mp_bt_zephyr_conn_t {
+    struct bt_conn *conn;
+    struct _mp_bt_zephyr_conn_t *next;
+} mp_bt_zephyr_conn_t;
+
 typedef struct _mp_bluetooth_zephyr_root_pointers_t {
+    // list of objects to be tracked by the gc
+    mp_obj_t objs_list;
```

Comment:
It might be better to use `m_tracked_calloc()` for this, rather than reimplement that functionality (unless you need specific functionality that `m_tracked_calloc()` is not providing).

Reply:
Thanks Damien. I see how `m_tracked_calloc()` works but it might be a bit of a hassle to free all the objects created during the service registrations (attributes, UUIDs, etc.) when de-initialising BLE, and with my approach I just have to make that `objs_list` variable point to NULL.